app/javascript/mastodon/features/ui/components/columns_area.js

```diff
@@ -74,9 +88,10 @@ export default class ColumnsArea extends ImmutablePureComponent {
 
   render () {
     const { columns, children, singleColumn } = this.props;
+    const { shouldAnimate } = this.state;
 
     const columnIndex = getIndex(this.context.router.history.location.pathname);
-    const shouldAnimate = Math.abs(this.lastIndex - columnIndex) === 1;
```

Comment:
When we use animation for location change based transition, `columnIndex` is already new one before transition, whereas it's old one on swiping based transition. So destination column will be rendered before animated transition.

I don't feel it improves perceived performance so much, on my (slow) device which takes some freeze time anyway. Also transition to rendering looks not bad, but different from behavior on swiping.

Well...I wonder that if we make same behavior as swiping, i.e. render destination while/after transition.